### ISSUE TITLE
Error when Compiling apk Magisk

### ISSUE THREAD

**Issue by Fays** [iBotPeaches/Apktool#1532]
### Information
1. **Apktool Version (`apktool -2.2.3`)** -
2. **Operating System (Linux)** -
3. **APK From? (XDA Dev)** -

Lien apk
https://mega.nz/#!cwIUXCjL!uANprWAnNIVQq1bbD68WHktvwTWOSJh_Z86u7xuXJOs

### Stacktrace/Logcat
```
java -jar apktool-2.2.3.jar b Magisk-5.0.2
I: Using Apktool 2.2.3-ea16f3-SNAPSHOT
I: Checking whether sources has changed...
I: Checking whether resources has changed...
I: Building resources...
W: res/drawable/$avd_hide_password__0.xml: Invalid file name: must contain only [a-zA-Z0-9_.]. Ignoring...
W: res/drawable/$avd_hide_password__1.xml: Invalid file name: must contain only [a-zA-Z0-9_.]. Ignoring...
W: res/drawable/$avd_hide_password__2.xml: Invalid file name: must contain only [a-zA-Z0-9_.]. Ignoring...
W: res/drawable/$avd_show_password__0.xml: Invalid file name: must contain only [a-zA-Z0-9_.]. Ignoring...
W: res/drawable/$avd_show_password__1.xml: Invalid file name: must contain only [a-zA-Z0-9_.]. Ignoring...
W: res/drawable/$avd_show_password__2.xml: Invalid file name: must contain only [a-zA-Z0-9_.]. Ignoring...
W: A/        ( 2176): First type is not attr!
Exception in thread "main" brut.androlib.AndrolibException: brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 134): [/tmp/brut_util_Jar_3024565972388081494.tmp, p, --forced-package-id, 127, --min-sdk-version, 21, --target-sdk-version, 26, --version-code, 42, --version-name, 5.0.2, --no-version-vectors, -F, /tmp/APKTOOL3219031163723745442.tmp, -0, arsc, -0, assets/private.key.pk8, -0, arsc, -I, /home/fayz/.local/share/apktool/framework/1.apk, -S, /home/fayz/APKS TRAD/Magisk-5.0.2/res, -M, /home/fayz/APKS TRAD/Magisk-5.0.2/AndroidManifest.xml]
	at brut.androlib.Androlib.buildResourcesFull(Androlib.java:477)
	at brut.androlib.Androlib.buildResources(Androlib.java:411)
	at brut.androlib.Androlib.build(Androlib.java:310)
	at brut.androlib.Androlib.build(Androlib.java:263)
	at brut.apktool.Main.cmdBuild(Main.java:231)
	at brut.apktool.Main.main(Main.java:84)
```

**Comment by iBotPeaches** [comment]
1. Invalid filenames, not sure why Magisk is using `$`. That is not valid. Guess I need to rip out the validation of file names as well.

2. The warning about `W: A/        ( 2176): First type is not attr!` is a duplicate of #1520, so I'll work on 1 and close this after that linking 2) as a duplicate.

**Comment by iBotPeaches** [comment]
I believe 1 is actually a supported feature in `aapt2`. See the recent comment here for details - https://github.com/iBotPeaches/Apktool/issues/1520#issuecomment-312359148

**Comment by iBotPeaches** [comment]
```
➜  Bug1532 apktool d com.topjohnwu.magisk_5.0.2-42_minAPI21\(armeabi-v7a,x86\)\(nodpi\)_apkmirror.com.apk 
I: Using Apktool 2.2.4-988fd1-SNAPSHOT on com.topjohnwu.magisk_5.0.2-42_minAPI21(armeabi-v7a,x86)(nodpi)_apkmirror.com.apk
I: Loading resource table...
I: Decoding AndroidManifest.xml with resources...
I: Loading resource table from file: /home/ibotpeaches/.local/share/apktool/framework/1.apk
I: Regular manifest package...
I: Decoding file-resources...
I: Decoding values */* XMLs...
I: Baksmaling classes.dex...
I: Copying assets and libs...
I: Copying unknown files...
I: Copying original files...
➜  Bug1532 apktool b com.topjohnwu.magisk_5.0.2-42_minAPI21\(armeabi-v7a,x86\)\(nodpi\)_apkmirror.com
I: Using Apktool 2.2.4-988fd1-SNAPSHOT
I: Checking whether sources has changed...
I: Smaling smali folder into classes.dex...
I: Checking whether resources has changed...
I: Building resources...
I: Copying libs... (/lib)
I: Building apk file...
I: Copying unknown files/dir...
➜  Bug1532 
```

This was fixed with the commit here: https://github.com/iBotPeaches/platform_frameworks_base/commit/3fe1781ece8b4d282f8edcc442b8e1734999c56b which was merged into apktool here: https://github.com/iBotPeaches/Apktool/commit/6ce02e143d79311bd396dd1d6613de1658048871

The other problems in this ticket were fixed with #1520
